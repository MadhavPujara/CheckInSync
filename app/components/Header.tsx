import React from "react";
import { Appbar } from "react-native-paper";

interface HeaderProps {
    onMenuPress: () => void;
}

export default function Header({ onMenuPress }: HeaderProps) {
    return (
        <Appbar.Header>
            <Appbar.Action icon="menu" onPress={onMenuPress} />
            <Appbar.Content title="Check-in Helper" />
        </Appbar.Header>
    );
}
