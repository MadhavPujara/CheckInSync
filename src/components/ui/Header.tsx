import React from "react";
import { Appbar } from "react-native-paper";

interface HeaderProps {
    onMenuPress: () => void;
}

export default function Header({ onMenuPress }: HeaderProps) {
    return (
        <Appbar.Header testID="header-container">
            <Appbar.Action
                testID="menu-button"
                icon="menu"
                onPress={onMenuPress}
            />
            <Appbar.Content testID="header-title" title="Check-in Helper" />
        </Appbar.Header>
    );
}
