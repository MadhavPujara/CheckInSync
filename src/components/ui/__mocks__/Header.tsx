import React from "react";
import { View, Text, TouchableOpacity } from "react-native";

interface HeaderProps {
    onMenuPress: () => void;
}

export default function MockHeader({ onMenuPress }: HeaderProps) {
    return (
        <View testID="header-container">
            <TouchableOpacity testID="menu-button" onPress={onMenuPress}>
                <Text>Menu</Text>
            </TouchableOpacity>
            <Text testID="header-title">Check-in Helper</Text>
        </View>
    );
}
