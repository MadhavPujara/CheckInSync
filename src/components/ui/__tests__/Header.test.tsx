import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import Header from "@/components/ui/Header";

// Mock react-native-safe-area-context
jest.mock("react-native-safe-area-context", () => ({
    useSafeAreaInsets: () => ({ top: 0, right: 0, bottom: 0, left: 0 }),
    SafeAreaProvider: ({ children }: { children: React.ReactNode }) => children,
}));

jest.mock("react-native-paper", () => {
    const { View, TouchableOpacity, Text } = require("react-native");
    return {
        Appbar: {
            Header: ({
                children,
                testID,
            }: {
                children: React.ReactNode;
                testID?: string;
            }) => <View testID={testID || "header-container"}>{children}</View>,
            Action: ({
                icon,
                onPress,
                testID,
            }: {
                icon: string;
                onPress: () => void;
                testID?: string;
            }) => (
                <TouchableOpacity testID={testID} onPress={onPress}>
                    <Text>{icon}</Text>
                </TouchableOpacity>
            ),
            Content: ({
                title,
                testID,
            }: {
                title: string;
                testID?: string;
            }) => <Text testID={testID}>{title}</Text>,
        },
    };
});

describe("Header", () => {
    it("renders correctly", () => {
        const mockOnMenuPress = jest.fn();
        const { getByTestId, getByText } = render(
            <Header onMenuPress={mockOnMenuPress} />
        );

        // Check if main components exist
        expect(getByTestId("header-container")).toBeTruthy();
        expect(getByTestId("menu-button")).toBeTruthy();
        expect(getByTestId("header-title")).toBeTruthy();

        // Check if title is shown
        expect(getByText("Check-in Helper")).toBeTruthy();
    });

    it("calls onMenuPress when menu button is pressed", () => {
        const mockOnMenuPress = jest.fn();
        const { getByTestId } = render(
            <Header onMenuPress={mockOnMenuPress} />
        );

        // Find and press the menu button
        const menuButton = getByTestId("menu-button");
        fireEvent.press(menuButton);

        // Verify callback was called
        expect(mockOnMenuPress).toHaveBeenCalledTimes(1);
    });

    it("matches snapshot", () => {
        const mockOnMenuPress = jest.fn();
        const { toJSON } = render(<Header onMenuPress={mockOnMenuPress} />);

        // Verify snapshot matches
        expect(toJSON()).toMatchSnapshot();
    });
});
