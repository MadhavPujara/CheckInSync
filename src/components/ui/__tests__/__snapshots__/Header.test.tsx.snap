// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<View
  testID="header-container"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="menu-button"
  >
    <Text>
      menu
    </Text>
  </View>
  <Text
    testID="header-title"
  >
    Check-in Helper
  </Text>
</View>
`;
